### PR TITLE
[WIP] Adding more dependency injection.

### PR DESCRIPTION
**Pull request by stevector**
Here's the d.o issue https://drupal.org/node/2281491

As of the creation of this pull request there is only one commit that adds a create method. It isn't adding any benefit yet and it is a work in progress.

### CODE REVIEW

**Comment by davereid** [comment]
So just an initial question - the plugin base doesn't actually use the entity manager anywhere. Should the individual plugins be responsible for pulling stuff from the container as they need them, or is the plugin base responsible? I have a feeling that the individual plugins should be responsible.

**Comment by stevector** [comment]
I really don't know. my thinking was "other child classes beside the default will need the entityManager" I'll look for a pattern elsewhere in core.

**Comment by stevector** [comment]
`ActionAddForm` makes me think you are correct. So does `CustomBlockController`. I'll switch it up.

**Comment by slashrsm** [comment]
I'd say individual plugins are responsible for things they need.

**Comment by stevector** [comment]
Thanks @slashrsm. I've pushed the refactor.

File: src/EntityEmbedDisplay/EntityEmbedDefaultDisplay.php [hunk 1]
```diff
@@ -17,7 +20,30 @@
  *
  * @todo Should this use a derivative? http://cgit.drupalcode.org/page_manager/tree/src/Plugin/Deriver/EntityViewDeriver.php
  */
-class EntityEmbedDefaultDisplay extends EntityEmbedDisplayBase {
+class EntityEmbedDefaultDisplay extends EntityEmbedDisplayBase implements ContainerFactoryPluginInterface {
+
```
Comment (davereid) [inline src/EntityEmbedDisplay/EntityEmbedDefaultDisplay.php:24]: I think we're missing a declaration for the $this->entityManager class property.

Copied from EntityEmbedFilter:

``` php
  /**
   * The entity manager service.
   *
   * @var \Drupal\Core\Entity\EntityManagerInterface
   */
  protected $entityManager;
```

Comment (stevector) [inline src/EntityEmbedDisplay/EntityEmbedDefaultDisplay.php:24]: Thanks! I just added it.

**Comment by davereid** [comment]
Great, looks good!

**Pull request merged**